proto/kvrpcpb.proto

```diff
@@ -655,3 +655,109 @@ message ReadIndexResponse{
     uint64 read_index = 2;
 }
 
+enum VerOp {
+    VerPut = 0;
+    VerDel = 1;
+}
+
+message VerMutation {
+    VerOp op = 1;
+    bytes key = 2;
+    bytes value = 3;
+}
+
+message VerValue {
+    bytes value = 1;
+    uint64 version = 2;
+}
+
+message VerKeyError {
+    string error = 1; 
+}
+
+message VerKvPair {
+    VerKeyError error = 1;
+    bytes key = 2;
+    repeated VerValue values = 3;
+}
+
+message VerGetRequest {
+    Context context = 1;
+    bytes key = 2;
+    string cf = 3; 
+    int32 version_num = 4;
+    uint64 start_version = 5; // start_version == 0 means without start version
+}
+
+message VerGetResponse {
+    errorpb.Error region_error = 1;
+    VerKeyError error = 2;
+    repeated VerValue value = 3;
+    bool not_found = 4;
+}
+
+message VerBatchGetRequest {
+    Context context = 1;
+    repeated bytes key = 2;
+    string cf = 3;
+    int32 version_num = 4;
+    uint64 start_version = 5;
+}
+
+message VerBatchGetResponse {
+    errorpb.Error region_error = 1;
+    repeated VerKvPair pairs = 2;
+}
+
+message VerMutRequest {
+    Context context = 1;
+    VerMutation mut = 2;
+    uint64 version = 3;
+    string cf = 4;
+}
+
+message VerMutResponse {
+    errorpb.Error region_error = 1;
+    VerKeyError error = 2;
+}
+
+message VerBatchMutRequest {
+    Context context = 1;
+    repeated VerMutation muts = 2;
+    uint64 version = 3;
+    string cf = 4;
+}
+
+message VerBatchMutResponse {
+    errorpb.Error region_error = 1;
+    VerKeyError error = 2;
+}
+
+message VerScanRequest {
+    Context context = 1;
+    bytes start_key = 2;
+    bytes end_key = 3;
+    string cf = 4;
+    uint32 limit = 5;
+    bool key_only = 6;
+    bool reverse = 7;
+    uint32 version_num = 8;
```

Comment:
what is it used for？

Reply:
here, `version_num` is same with that in `VerGet`, means a specified number of versions to scan.